core/src/main/java/edu/ucr/cs/riple/core/Annotator.java

```diff
@@ -216,4 +224,71 @@ private ImmutableSet<Report> processTriggeredFixes(
     // Result of the iteration analysis.
     return explorer.explore();
   }
+
+  /**
+   * Resolves all remaining errors in target module by following steps below:
+   *
+   * <ul>
+   *   <li>Enclosing method of triggered errors will be marked with {@code @NullUnmarked}
+   *       annotation.
+   *   <li>Uninitialized fields (inline or by constructor) will be annotated as
+   *       {@code @SuppressWarnings("NullAway.init")}.
+   *   <li>Explicit {@code Nullable} assignments to fields will be annotated as
+   *       {@code @SuppressWarnings("NullAway")}.
+   * </ul>
+   *
+   * @param fieldDeclarationAnalysis Field declaration analysis.
+   * @param tree Method Declaration analysis.
+   */
+  private void forceResolveRemainingErrors(
+      FieldDeclarationAnalysis fieldDeclarationAnalysis, MethodDeclarationTree tree) {
+    // Collect regions with remaining errors.
+    Utility.buildTarget(config);
+    List<Error> remainingErrors = Utility.readErrorsFromOutputDirectory(config.target);
+    Set<Fix> remainingFixes =
+        Utility.readFixesFromOutputDirectory(
+            config.target, Fix.factory(config, fieldDeclarationAnalysis));
+
+    // Collect all regions for NullUnmarked.
```

Comment:
Probably worth adding here some explanation of which regions qualify for NullUnmarked. Maybe:

```
// For all errors in regions which correspond to a method's body, we can add @NullUnmarked at the method level.
```

?

p.s. presumably, in the future, it might make sense to add `@NullUnmarked` to classes (e.g. inner classes) in specific circumstances, but for now the above seems like a correct characterization. No?

Reply:
559b012597d457467dd96aa995a4947f3faf61a3

I updated the comment, Yes the logic seems correct to me. Sure in future we can have follow ups for marking a class as `@NullUnmarked` instead of marking most of its methods.